samza-core/src/main/java/org/apache/samza/serializers/model/JsonTaskModelMixIn.java

```diff
@@ -45,4 +46,7 @@ public JsonTaskModelMixIn(@JsonProperty("task-name") TaskName taskName, @JsonPro
 
   @JsonProperty("changelog-partition")
   abstract Partition getChangelogPartition();
+
+  @JsonProperty("task-mode")
```

Comment:
Do you need to do anything to mark this as optional or will this work as-is if the property is missing?

Reply:
It'll be ok asis because at the class level we have @JsonIgnoreProperties(ignoreUnknown = true). 
It'll set it to java default, and my other PR will handle the case in code.

Reply:
Thanks. Added deserializer for handling this case. 
Also added a test for this case in TestSamzaObjectMapper.